### PR TITLE
Support system-wide Dark Theme/Dark Mode in android 10; update AppCompat support lib to androidx

### PR DESCRIPTION
**Pull request by jdmonin**
I've updated Notepad to support Android 10's system-wide Dark Mode. `NoteEditor` wanted extra attention because of notepad theme colors.

To do so, I'm using DayNight from androidX AppCompat. So, I updated the code from  using the v4 support library to androidx.

Also raised minSdkVersion to 14 for androidX appcompat requirements, and now using gradle 5.

Tested on:
- Android 5.0.2 device (doesn't support Dark Theme; no regressions seen)
- Android 9 emulator, controlling Dark Theme through developer options
- Android 10 device

Please let me know if you need anything else to merge this request.

Thank you!

### CODE REVIEW

File: NotePad/src/main/java/org/openintents/notepad/NoteEditor.java [hunk 1]
```diff
@@ -498,6 +498,17 @@ protected void onCreate(Bundle savedInstanceState) {
         // res/layout/note_editor.xml
         setContentView(R.layout.note_editor);
 
+        // now that we have the layout, finish actionbar setup
+        if (mActionBarAvailable) {
+            setSupportActionBar((Toolbar) findViewById(R.id.oi_toolbar));
+            WrapActionBar bar = new WrapActionBar(this);
+            bar.setDisplayHomeAsUpEnabled(true);
+            // force to show the actionbar on version 14+
+            if (Integer.valueOf(android.os.Build.VERSION.SDK) >= 14) {
```
Comment (friedger) [inline NotePad/src/main/java/org/openintents/notepad/NoteEditor.java:507]: This is true for all builds as the minSdk is 14, furthermore android.os.Build.VERSION.SDK_INT should be used.

File: NotePad/src/main/java/org/openintents/notepad/noteslist/NotesList.java [hunk 2]
```diff
@@ -599,47 +607,9 @@ public boolean onCreateOptionsMenu(Menu menu) {
...
-
-            // Give a shortcut to the edit action.
-            if (items[0] != null) {
-                items[0].setShortcut('1', 'e');
-            }
-        } else {
-            menu.removeGroup(Menu.CATEGORY_ALTERNATIVE);
-        }
+        // Skip any item-specific actions: those go in context menu, not options menu
+        menu.removeGroup(Menu.CATEGORY_ALTERNATIVE);
```
Comment (friedger) [inline NotePad/src/main/java/org/openintents/notepad/noteslist/NotesList.java:612]: Why did you remove the alternative menu items?

File: NotePad/src/main/java/org/openintents/notepad/NoteEditor.java (hunk 1 above)
Comment (jdmonin) [reply NotePad/src/main/java/org/openintents/notepad/NoteEditor.java:507]: Thank you, good point. I've cleaned this up now.

File: NotePad/src/main/java/org/openintents/notepad/noteslist/NotesList.java (hunk 2 above)
Comment (jdmonin) [reply NotePad/src/main/java/org/openintents/notepad/noteslist/NotesList.java:612]: Apologies, I removed too much here as part of my conversion from ListActivity.
This has now been restored, based on the code in master.
I validated restoration by running  
`git diff master...HEAD  NotePad/src/main/java/org/openintents/notepad/noteslist/NotesList.java`

I wasn't able to find out much about what the alternative items would do, so I don't have a good way to completely test this. I only did a quick test in the emulator to make sure the context menu appears.

**Comment by friedger** [comment]
Thank you for the contribution!

**Comment by jdmonin** [comment]
You're welcome, thank you for reviewing it!
